ProjectWar/GameException.h

```diff
@@ -33,5 +33,20 @@ class InvalidPositionException : public std::runtime_error {
     int width;
     int height;
 };
+        
+class IllegalStateException : public std::runtime_error {
+public:
+    IllegalStateException( std::string msg )
```

Comment:
In general is a good practice when you are passing a single instance in order to avoid one allocation. Only for the build types like int, short or char it doesn´t make that much of a difference.